core/src/mindustry/world/blocks/power/ItemLiquidGenerator.java

```diff
@@ -134,7 +134,7 @@ public void updateTile(){
                     Item item = items.take();
                     productionEfficiency = getItemEfficiency(item);
                     explosiveness = item.explosiveness;
-                    generateTime = 1f;
+                    generateTime %= 1f;
```

Comment:
yikes

Reply:
-0.3 % 1 should = 0.7.
Or should I convert to `Mathf.mod` instead?

Reply:
> -0.3 % 1 should = 0.7.

No, that's not how mod works, and there's no reason to use mod here at all.